src/core/load-fragment.c

```diff
@@ -1623,6 +1623,22 @@ int config_parse_exec_cpu_sched_prio(
         return 0;
 }
 
+static int mount_options_set_and_consume(MountOptions **options, PartitionDesignator d, char **s) {
+        assert(options);
+        assert(d >= 0);
+        assert(s);
+
+        if (!*options) {
+                *options = new0(MountOptions, 1);
+                if (!*options)
+                        return log_oom();
+        }
+
+        free_and_replace((*options)->options[d], *s);
+
+        return 0;
```

Comment:
ah, look, you lready have something so close to what i suggested above, maybe move to generic code, and offer a non-consuming variant, and use it everyhwere.